compiler/fir/raw-fir/fir-common/src/org/jetbrains/kotlin/fir/builder/BaseFirBuilder.kt

```diff
@@ -722,8 +731,243 @@ abstract class BaseFirBuilder<T>(val baseSession: FirSession, val context: Conte
                         }
                     }
 
+                    // TODO: copy should call <init> with parameters and return the copied instance.
```

Comment:
This seems like a coverage hole. A simple test, like copy() and test ===, equals, hashCode, and toString would be good enough. I can do that in a follow-up PR.

Reply:
It turns out there are couple BB tests that actually cover data class copy, followed by other synthetic member calls.